x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/RestVectorTileAction.java

```diff
@@ -75,6 +77,8 @@
     // mime type as defined by the mapbox vector tile specification
     private static final String MIME_TYPE = "application/vnd.mapbox-vector-tile";
 
+    private static final String CENTROID_AGG_NAME = "[centroid]";
+
```

Comment:
It is not allowed to use '[' or ']' in the name of aggregations using the rest interface so calling like this prevents name collision.

Reply:
That looks a bit hack-y. I wonder if we would be better off by adding a special prefix to all internal aggs and prohibiting external aggs to use this prefix in their names?

Reply:
I agree, it looks like a bug that the rest interface does not allow that construct but you do it in java. 

My first idea was to add a prefix to the user defined aggregations so we do not need to add any restriction but it seems that would be pretty invasive as at the moment there is no tooling for doing that. 

We can add a prefix to the internal aggs, let say they should start with an underscore '_' and the API fails if a user sends an aggregation which name starts with that prefix. @thomasneirynck would that be an issue for you?

Reply:
I think if we pick `_mvt_` or something like that, it would have much less chances to conflict with anything.